qiskit/chemistry/algorithms/minimum_eigen_solvers/oovqe.py

```diff
@@ -0,0 +1,604 @@
+        if qmolecule.mo_coeff_b is not None:
+            mo_eri_b = qmolecule.twoeints2mo(qmolecule.eri,
+                                             qmolecule.mo_coeff_b)
+            norbs = qmolecule.mo_coeff.shape[0]
+            qmolecule.mo_eri_ints_bb = mo_eri_b.reshape(norbs, norbs, norbs, norbs)
+            qmolecule.mo_eri_ints_ba = qmolecule.twoeints2mo_general(
+                qmolecule.mo_coeff_b, qmolecule.mo_coeff_b, qmolecule.mo_coeff,
+                qmolecule.mo_coeff)
+            qmolecule.mo_eri_ints_ba = qmolecule.mo_eri_ints_ba.reshape(norbs, norbs,
+                                                                        norbs, norbs)
+        # dipole integrals (if available) from AO to MO
+        if qmolecule.x_dip_ints is not None:
+            qmolecule.x_dip_mo_ints = qmolecule.oneeints2mo(qmolecule.x_dip_ints,
+                                                            qmolecule.mo_coeff)
+            qmolecule.y_dip_mo_ints = qmolecule.oneeints2mo(qmolecule.y_dip_ints,
+                                                            qmolecule.mo_coeff)
+            qmolecule.z_dip_mo_ints = qmolecule.oneeints2mo(qmolecule.z_dip_ints,
+                                                            qmolecule.mo_coeff)
+        # support for unrestricted spins
+        if qmolecule.mo_coeff_b is not None and qmolecule.x_dip_ints is not None:
+            qmolecule.x_dip_mo_ints_b = qmolecule.oneeints2mo(qmolecule.x_dip_ints,
+                                                              qmolecule.mo_coeff_b)
+            qmolecule.y_dip_mo_ints_b = qmolecule.oneeints2mo(qmolecule.y_dip_ints,
+                                                              qmolecule.mo_coeff_b)
+            qmolecule.z_dip_mo_ints_b = qmolecule.oneeints2mo(qmolecule.z_dip_ints,
+                                                              qmolecule.mo_coeff_b)
+
+
+class OrbitalRotation:
+    r"""
+    Class that regroups methods for creation of matrices that rotate the MOs.
+    It allows to create the unitary matrix U = exp(-kappa) that is parameterized with kappa's
+    elements. The parameters are the off-diagonal elements of the anti-hermitian matrix kappa.
+    """
+
+    def __init__(self,
+                 num_qubits: int,
+                 core: Optional[LegacyBaseOperator] = None,
+                 qmolecule: Optional[QMolecule] = None,
+                 orbital_rotations: list = None,
+                 orbital_rotations_beta: list = None,
+                 parameters: list = None,
+                 parameter_bounds: list = None,
+                 parameter_initial_value: float = 0.1,
+                 parameter_bound_value: tuple = (-2 * np.pi, 2 * np.pi)) -> None:
+
+        """
+        Args:
+            num_qubits: number of qubits necessary to simulate a particular system.
+            core: instance of the :class:`~qiskit.chemistry.core.Hamiltonian` class to
```

Comment:
It seems this is passed into here so as to be able to access info around frozen core. Now the core.Hamiltonian allows the arbitrary freezing/removal of orbitals? Removal most often being just done with LiH since in general it can affect the outcome too much. I see some special treatment around frozen so just wondering what sort of operators are compatible given the parameters core.Hamiltonian has for computing them. Presumably any mapping is fine, Z2Symettries too? but not sure about the orbital freezing/removal due to the special treatment here